### PR TITLE
Add `--include` and `--exclude` filters to the `ls` and `rm` commands

### PR DESCRIPTION
**Pull request by emnoor-reef**

### CODE REVIEW

File: b2/_internal/_utils/uri.py [hunk 1]
```diff
@@ -165,10 +165,10 @@ def list_file_versions_by_uri(self, uri, *args, **kwargs):
         raise NotImplementedError(f"Unsupported URI type: {type(uri)}")
 
     @list_file_versions_by_uri.register
-    def _(self, uri: B2URI, *args, **kwargs):
+    def _(self, uri: B2URI, *args, filters=(), **kwargs):
```
Comment (agoncharov-reef) [inline b2/_internal/_utils/uri.py:168]: Type hint?

File: b2/_internal/console_tool.py [hunk 2]
```diff
@@ -2175,6 +2176,8 @@ def _setup_parser(cls, parser):
         parser.add_argument('--versions', action='store_true')
         parser.add_argument('-r', '--recursive', action='store_true')
         parser.add_argument('--withWildcard', action='store_true')
+        parser.add_argument('--include', dest='filters', action='append', type=Filter.include)
+        parser.add_argument('--exclude', dest='filters', action='append', type=Filter.exclude)
```
Comment (agoncharov-reef) [inline b2/_internal/console_tool.py:2180]: The docstring is outdated now.

File: b2/_internal/console_tool.py [hunk 3]
```diff
@@ -2198,6 +2201,7 @@ def _get_ls_generator(self, args):
             latest_only=not args.versions,
             recursive=args.recursive,
             with_wildcard=args.withWildcard,
+            filters=args.filters or (),
```
Comment (agoncharov-reef) [inline b2/_internal/console_tool.py:2204]: Maybe use default value in `add_argument` instead? Is this possible?
```python
parser.add_argument('--include', dest='filters', action='append', type=Filter.include, default=())
```

File: test/unit/test_console_tool.py [hunk 4]
```diff
@@ -2445,6 +2445,64 @@ def test_ls_b2id(self):
                 '''
         self._run_command(['ls', f'b2id://{file_version.id_}'], expected_stdout, '', 0)
 
+    def test_ls_filters(self):
```
Comment (agoncharov-reef) [inline test/unit/test_console_tool.py:2448]: This checks filters working on `ls` command but doesn't ensure they work with `rm` command. I understand that it should work automatically bc of some inheritance, but still I think some very basic test would be good here. If some code changes in future, we don't want `rm` to misbehave with those flags.

File: b2/_internal/console_tool.py (hunk 3 above)
Comment (emnoor-reef) [reply b2/_internal/console_tool.py:2204]: That doesn't work. With `action='append'`, argparse stores the default, then tries to call `.append()` on that. So tuples don't work here.

But we can have `default=[]` here. A quick forage through argparse code verifies that having multiple flags with dest='filters' and a default won't cause any problems.

**Review by mjurbanski-reef — APPROVED** [review]

**Pull request merged**